src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -93,7 +93,11 @@ public static IEnumerable<VisualStudioInstance> QueryVisualStudioInstances()
         /// <returns>Instance of Visual Studio found and registered.</returns>
         public static VisualStudioInstance RegisterDefaults()
```

Comment:
I was trying to implement https://github.com/microsoft/MSBuildLocator/pull/106#discussion_r542216240

Is that wrong?

Reply:
Ah. Yes, I now think it's not the right design. We could potentially add another overload that takes a bool to "give me everything, I don't care" but I don't think we should filter in RegisterDefaults and I do think we should present only usable instances in the list by default.

Reply:
Reverted to filtering in DotNetSdkLocationHelper.cs.

Reply:
I'm a little resistant to add an overload for "give me everything" without anyone explicitly asking for it. (I'm guessing most people who use MSBuildLocator use it for relatively straightforward purposes.) It would also make it harder to pull in my unit testing PR when that comes along.

Reply:
How would this affect a unit testing PR?

Reply:
Two reasons: it would add something else I'd probably want to test, and since I'm already planning to add an overload to this, I'd have to split my overload into "give me everything" and "don't give me everything" cases. Not a huge deal, though.